File: Csharp/ChesszumSpass/Drawer.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChesszumSpass
{
    class Drawer
    {
        Bitmap bitmap;
        int size;

        Image[] pieceimg;

        public Drawer(int size)
        {
            this.size = size;
            bitmap = new Bitmap(size, size);

            pieceimg = new Bitmap[12];
            for (int i = 0; i < 12; i++)
            {
                pieceimg[i] = Bitmap.FromFile("..\\images\\" + ((Pieces)(i + 1)).ToString() + ".png");
            }
        }

        public Bitmap update(Game game)
        {
            int rectsize = size / 8;
            int field = 0;
            using (Graphics g = Graphics.FromImage(bitmap))
            {
                for (int y = 0; y < 8; y++)
                {
                    for (int x = 0; x < 8; x++)
                    {
                        bool movpos = false;
                        if (game.ismarked)
                        {
                            for (int i = 0; i < game.possiblePos.Count; i++)
                            {
                                if (((game.whiteTurn && game.possiblePos[i].x == x) || (!game.whiteTurn && game.possiblePos[i].x == 7 - x)) && ((game.whiteTurn && game.possiblePos[i].y == y) || (!game.whiteTurn && game.possiblePos[i].y == 7 - y)))
                                {
                                    movpos = true;
                                    break;
                                }
                            }
                        }
                        if (movpos)
                        {
                            if (field % 2 == 0)
                            {
                                g.FillRectangle(Brushes.LightGray, rectsize * x, rectsize * y, rectsize, rectsize);
                            }
                            else
                            {
                                g.FillRectangle(Brushes.SlateGray, rectsize * x, rectsize * y, rectsize, rectsize);
                            }
                        }
                        else
                        {
                            if (field % 2 == 0)
                            {
                                g.FillRectangle(Brushes.RosyBrown, rectsize * x, rectsize * y, rectsize, rectsize);
                            }
                            else
                            {
                                g.FillRectangle(Brushes.Brown, rectsize * x, rectsize * y, rectsize, rectsize);
                            }
                        }
                        field++;
                    }
                    field++;
                }

                Font font = new Font(FontFamily.GenericMonospace, (float)(rectsize / 8));
                for (int i = 0; i < 8; i++)
                {
                    int pos = 8 - i;
                    if (!game.whiteTurn) pos = i + 1;
                    g.DrawString(pos.ToString(), font, Brushes.Black, 1 , 1 + rectsize * i);
                    g.DrawString(Convert.ToChar(Convert.ToByte('a') + 8 - pos).ToString(), font, Brushes.Black, 1 + rectsize * i + (float)(rectsize * 0.8), 1 + 7 * rectsize + (float)(rectsize * 0.8));
                }

                for (int y = 0; y < 8; y++)
                {
                    for (int x = 0; x < 8; x++)
                    {
                        if (game.board.pieces[y, x] != Pieces.EMPTY)
                        {
                            int posx = rectsize * x;
                            int posy = rectsize * y;
                            if (!game.whiteTurn)
                            {
                                posx = rectsize * (7 - x);
                                posy = rectsize * (7 - y);
                            }
                            g.DrawImage(pieceimg[((int)game.board.pieces[y, x]) - 1], posx, posy, rectsize, rectsize);
                        }
                    }
                }

                if (game.ismarked && game.marked.x >= 0 && game.marked.y >= 0 && game.marked.x < 8 && game.marked.y < 8)
                {
                    int posx = game.marked.x;
                    int posy = game.marked.y;
                    if (!game.whiteTurn)
                    {
                        posx = 7 - posx;
                        posy = 7 - posy;
                    }
                    g.FillRectangle(Brushes.LimeGreen, rectsize * posx, rectsize * posy, rectsize, rectsize / 16);
                    g.FillRectangle(Brushes.LimeGreen, rectsize * posx, rectsize * posy + rectsize / 16 * 15, rectsize, rectsize / 16);
                    g.FillRectangle(Brushes.LimeGreen, rectsize * posx, rectsize * posy, rectsize / 16, rectsize);
                    g.FillRectangle(Brushes.LimeGreen, rectsize * posx + rectsize / 16 * 15, rectsize * posy, rectsize / 16, rectsize);
                }
            }

            return bitmap;
        }
    }
}

File: HTML5/drawer.js
var drawer = new Drawer();

function Drawer()
{
   this.draw = function(x, y, sizeX, sizeY)
   {  
      var height = canvas.height - 50;
      var width = canvas.width;
      if (height > width) height = width;
      if (width > height) width = height;   
      var rectsize = Math.floor(height / 8);

      var field = 0;
      for (var y = 0; y < 8; y++)
      {
         for (var x = 0; x < 8; x++)
         {
            var movpos = false;
            if (game.ismarked)
            {
               for (var i = 0; i < game.possiblePos.length; i++)
               {
                  if (((settings.playerWhite && game.possiblePos[i].x == x) || (!settings.playerWhite && game.possiblePos[i].x == 7 - x)) && ((settings.playerWhite && game.possiblePos[i].y == y) || (!settings.playerWhite && game.possiblePos[i].y == 7 - y)))
                  {
                     movpos = true;
                     break;
                  }
               }
            }
            if (field % 2 == 0)
            {
               canvascontext.drawImage(Images.TILE1.image, rectsize * x,50 + rectsize * y, rectsize, rectsize);
            }
            else
            {
               canvascontext.drawImage(Images.TILE0.image, rectsize * x,50 + rectsize * y, rectsize, rectsize);
            }
            if (movpos)
            {
               if (field % 2 == 0)
               {
                  canvascontext.fillStyle = `rgb(153, 230, 153)`;
               }
               else
               {
                  canvascontext.fillStyle = `rgb(45, 185, 45)`;
               }
               canvascontext.fillRect(rectsize * x,50 + rectsize * y,rectsize, rectsize);
            }
            field++;
         }
         field++;
      }
      
      for (var i = 0; i < 8; i++)
      {
         var pos = 8 - i;
         if (!settings.playerWhite) pos = i + 1;
         canvascontext.fillStyle = `rgb(0, 0, 0)`;
         textblock.draw(0, 50 + rectsize * i, 50, 20, "black", "" + pos);
         textblock.draw(rectsize * i + Math.floor(rectsize * 0.85), 50 + 7 * rectsize + Math.floor(rectsize * 0.8), 50, 20, "black", String.fromCharCode(("a".charCodeAt(0)) + 8 - pos));
      }
      
      for (var y = 0; y < 8; y++)
      {
         for (var x = 0; x < 8; x++)
         {
            if (board.pieces[y][x] != Pieces.EMPTY)
            {
               var posx = rectsize * x;
               var posy = rectsize * y;
               if (!settings.playerWhite) 
               {
                  posx = rectsize * (7 - x);
                  posy = rectsize * (7 - y);
               }
               var imagepos = board.pieces[y][x] - 1;
               canvascontext.drawImage(Imagelist[imagepos].image, posx, 50 + posy, rectsize, rectsize);
            }
         }
      }
      
      if (game.ismarked && game.marked.x >= 0 && game.marked.y >= 0 && game.marked.x < 8 && game.marked.y < 8)
      {
          var posx = game.marked.x;
          var posy = game.marked.y;
          if (!settings.playerWhite) 
          {
             posx = 7 - posx;
             posy = 7 - posy;
          }
          canvascontext.fillStyle = `rgb(255, 0, 0)`;
          canvascontext.fillRect(rectsize * posx, 50 + rectsize * posy, rectsize, rectsize / 16);
          canvascontext.fillRect(rectsize * posx, 50 + rectsize * posy + rectsize / 16 * 15, rectsize, rectsize / 16);
          canvascontext.fillRect(rectsize * posx, 50 + rectsize * posy, rectsize / 16, rectsize);
          canvascontext.fillRect(rectsize * posx + rectsize / 16 * 15, 50 + rectsize * posy, rectsize / 16, rectsize);
      }
      
   }
  
} 
